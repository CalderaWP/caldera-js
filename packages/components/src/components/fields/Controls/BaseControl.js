import React from 'react';
import {FieldWrapper} from "../../..";
import {FieldLabel} from "../FieldLabel/FieldLabel";
import {labelClassNames} from "../util";

function BaseControl( { id, label, help, className, children,fieldType } ) {
	return (
		<FieldWrapper
			fieldType={fieldType}>
			<FieldLabel
				fieldId={id}
			>
				{label}
			</FieldLabel>
			{children}
			{ !! help && <p id={ id + '__help' } className="caldera-control__help description">{ help }</p> }
		</FieldWrapper>
	);
}

export default BaseControl;
