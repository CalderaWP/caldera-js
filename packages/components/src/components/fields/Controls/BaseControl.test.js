import * as React from 'react';
import renderer from 'react-test-renderer';

import  BaseControl  from './BaseControl';
//import { mount } from 'enzyme';

describe('BaseControl ', () => {
	let onChange;
	let onBlur;

	beforeEach(() => {
		onChange = jest.fn();
		onBlur = jest.fn();
	});

	it('Renders children', () => {
		const component = renderer.create(
			<BaseControl
				id={'a1'}
				label={'label'}
				help={'help-text'}
				fieldType={'text'}
				onChange={onChange}
				onBlur={onBlur}
			>
				<input/>
			</BaseControl>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Puts label after', () => {
		const component = renderer.create(
			<BaseControl
				id={'a1'}
				label={'label'}
				help={'help-text'}
				fieldType={'text'}
				onChange={onChange}
				onBlur={onBlur}
				labelBefore={false}
			>
				<input/>
			</BaseControl>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Renders the messages if passes', () => {
		const MessageZone = () => (<div className={'test-find'}>Hi Roy</div>);

		const component = renderer.create(
			<BaseControl
				id={'a1'}
				label={'label'}
				help={'help-text'}
				fieldType={'text'}
				onChange={onChange}
				onBlur={onBlur}
				labelBefore={false}
				Messages={MessageZone}
			>
				<input/>
			</BaseControl>
		);
		const testInstance = component.root;
		expect(testInstance.findByProps({className: "test-find"}).children).toEqual(['Hi Roy']);


	});

});
