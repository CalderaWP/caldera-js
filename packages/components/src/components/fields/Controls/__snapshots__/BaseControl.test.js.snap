// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseControl  Puts label after 1`] = `
<div
  className="caldera-field-wrapper caldera-field-wrapper-text"
>
  <input />
  <label
    className="caldera-field-label caldera-field-label-text"
    htmlFor="a1"
  >
    label
     
  </label>
  <p
    className="caldera-control__help description"
    id="a1__help"
  >
    help-text
  </p>
</div>
`;

exports[`BaseControl  Puts label after 2`] = `
<div
  className="caldera-field-wrapper caldera-field-wrapper-text"
>
  <input />
  <label
    className="caldera-field-label caldera-field-label-text"
    htmlFor="a1"
  >
    label
     
  </label>
  <p
    className="caldera-control__help description"
    id="a1__help"
  >
    help-text
  </p>
</div>
`;

exports[`BaseControl  Renders children 1`] = `
<div
  className="caldera-field-wrapper caldera-field-wrapper-text"
>
  <label
    className="caldera-field-label caldera-field-label-text"
    htmlFor="a1"
  >
    label
     
  </label>
  <input />
  <p
    className="caldera-control__help description"
    id="a1__help"
  >
    help-text
  </p>
</div>
`;
